Bearded.UI/EventArgs/ModifierKeys.cs

```diff
@@ -26,5 +29,70 @@ public static ModifierKeys FromInputManager(InputManager inputManager)
                 inputManager.IsKeyPressed(Key.LAlt) || inputManager.IsKeyPressed(Key.RAlt),
                 inputManager.IsKeyPressed(Key.LWin) || inputManager.IsKeyPressed(Key.RWin));
         }
+
+        public static Builder GetBuilder() => new Builder();
+
+        public sealed class Builder
```

Comment:
The surrounding type is a struct, but this one is a class? That doesn't seem right to me at all. Do we need the builder at all? What's the use case for it? Isn't a static construction method enough?

Alternatively, why not have `WithShift()` etc. methods on the original struct, which would return a new copy with the changes made?

Reply:
> The surrounding type is a struct, but this one is a class?

Since the builder is mutable. If you start passing that around, but it's a struct, weird things happen. The surrounding type was originally a class as well, but that was changed based on an earlier review comment.

> Do we need the builder at all? What's the use case for it? Isn't a static construction method enough?

A static constructor method for a class that basically takes a list of booleans in? That doesn't sound very friendly. I think this goes beyond the scope of a static constructor method, given that all the different parameters can be confused with each other. You could use named parameters, but I'd rather enforce it through the use of a builder.

> Alternatively, why not have WithShift() etc. methods on the original struct, which would return a new copy with the changes made?

I do not want to imply that the actual `ModifierKeys` object passed around can be changed.

Reply:
> I do not want to imply that the actual `ModifierKeys` object passed around can be changed.

How would that be implied with such methods?

I'm fine not having static constructors (Though you could use a flag enum instead of bools for that if you prefer). However, the builder is really weird to me. Why do we need to pass it around? Do you have an example use case? I can't wrap my head around that.

If you prefer having a builder just to create instances locally, you could still make it a readonly struct and use chaining methods like I suggest, though then I'd question why you don't just add them to `ModifierKeys` as I suggested. Or does that have some downside I don't understand?

Reply:
It's been too long for me to remember why. Replaced it with methods that allow you to chain together modifier keys for construction as suggested. Not sure if we also need "without" methods, but we can always add those when needed.